macos/FluentUI/Extensions/Color+Apperance.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import Foundation
+import AppKit
+
+private let kAppleInterfaceStyle = "AppleInterfaceStyle"
+
+public extension NSAppearance {
+
+	/// Pseudo algorithm picked up from https://developer.apple.com/forums/thread/118974
+	var isDarkMode: Bool {
+		if #available(OSX 10.15, *) {
+			let appearanceDescription = NSApplication.shared.effectiveAppearance.debugDescription.lowercased()
+			if appearanceDescription.contains("dark") {
+				return true
+			}
+		} else if #available(OSX 10.14, *) {
+			if let appleInterfaceStyle = UserDefaults.standard.object(forKey: kAppleInterfaceStyle) as? String {
+				if appleInterfaceStyle.lowercased().contains("dark") {
+					return true
+				}
+			}
+		}
+		return false
+	}
+}
+
+@objc(MSFDynamicColor)
+public class DynamicColor: NSObject {
+
+	private let background: NSColor
```

Comment:
These properties should be light and dark instead of background and foreground.
And then you have just resolvedColor as a method.